src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -140,6 +140,14 @@ bool CanConsumeRowNonSequentially
         /// </summary>
         char[]? _tempCharBuf;
 
+        /// <summary>
+        /// Used to keep track of every unique row this reader object ever traverses.
+        /// This is used to detect whether nested DbDataReaders are still valid.
+        /// </summary>
+        internal ulong UniqueRowId;
+
+        internal NpgsqlNestedDataReader? CachedFreeNestedDataReader;
```

Comment:
nit: CachedNestedDataReader?

Reply:
I thought it was a good idea to signify that the reader is free.

Reply:
OK by me